llvm/test/SYCLLowerIR/ConditionalCallOnDevice/conditional-call-on-device-2.ll

```diff
@@ -0,0 +1,61 @@
+entry:
+  %agg.tmp = alloca %class.anon, align 8
+  %fn.ascast = addrspacecast ptr %fn to ptr addrspace(4)
+  call spir_func void @_ZN4sycl3_V13ext6oneapi12experimental6detail31call_if_on_device_conditionally1(ptr noundef byval(%class.anon) align 8 %agg.tmp, i32 noundef -2, i32 noundef 251660032) #9
+  ret void
+}
+
+; CHECK-NOT: call spir_func void @_ZN4sycl3_V13ext6oneapi12experimental6detail31call_if_on_device_conditionally1(
+; CHECK: call spir_func void @_ZN4sycl3_V13ext6oneapi12experimental6detail31call_if_on_device_conditionally1_PREFIX_1(ptr @CallableFunc, ptr %agg.tmp, i32 -2, i32 251660032)
+; CHECK-NOT: call spir_func void @_ZN4sycl3_V13ext6oneapi12experimental6detail31call_if_on_device_conditionally1(
+
+define internal spir_func void @_ZN4sycl3_V13ext6oneapi12experimental6detail38call_if_on_device_conditionally_helper2(ptr noundef byval(%class.anon) align 8 %fn, ptr noundef byval(%"struct.std::integer_sequence.3") align 1 %0) #2 !srcloc !0 {
+entry:
+  %agg.tmp = alloca %class.anon, align 8
+  %fn.ascast = addrspacecast ptr %fn to ptr addrspace(4)
+  call spir_func void @_ZN4sycl3_V13ext6oneapi12experimental6detail31call_if_on_device_conditionally2(ptr noundef byval(%class.anon) align 8 %agg.tmp, i32 noundef -2, i32 noundef 251660032) #9
+  ret void
+}
+
+; CHECK-NOT: call spir_func void @_ZN4sycl3_V13ext6oneapi12experimental6detail31call_if_on_device_conditionally2(
+; CHECK: call spir_func void @_ZN4sycl3_V13ext6oneapi12experimental6detail31call_if_on_device_conditionally2_PREFIX_2(ptr @CallableFunc, ptr %agg.tmp, i32 -2, i32 251660032)
+; CHECK-NOT: call spir_func void @_ZN4sycl3_V13ext6oneapi12experimental6detail31call_if_on_device_conditionally2(
+
+define internal spir_func void @_ZN4sycl3_V13ext6oneapi12experimental6detail31call_if_on_device_conditionally1(ptr noundef byval(%class.anon) align 8 %fn, i32 noundef %0, i32 noundef %1) #7 !srcloc !1 {
+entry:
+  %fn.ascast = addrspacecast ptr %fn to ptr addrspace(4)
+  call spir_func void @CallableFunc(ptr addrspace(4) noundef align 8 dereferenceable_or_null(8) %fn.ascast) #9
+  ret void
+}
+
+define internal spir_func void @_ZN4sycl3_V13ext6oneapi12experimental6detail31call_if_on_device_conditionally2(ptr noundef byval(%class.anon) align 8 %fn, i32 noundef %0, i32 noundef %1) #7 !srcloc !1 {
+entry:
+  %fn.ascast = addrspacecast ptr %fn to ptr addrspace(4)
+  call spir_func void @CallableFunc(ptr addrspace(4) noundef align 8 dereferenceable_or_null(8) %fn.ascast) #9
+  ret void
+}
+
+define internal spir_func void @CallableFunc(ptr addrspace(4) noundef align 8 dereferenceable_or_null(8) %this) #6 align 2 !srcloc !2 {
+entry:
+  ret void
+}
+
+; CHECK-NOT: define internal spir_func void @_ZN4sycl3_V13ext6oneapi12experimental6detail31call_if_on_device_conditionally1(
```

Comment:
If you don't expect this function to be present in a module at all, then you can just use `--implicit-check-not @namegoeshere`. It will cover both `define` and `call` checks. The option can be passed multiple times to specify different patterns to look for.